documentation/src/main/wiki/BestPractices.md

```diff
@@ -0,0 +1,218 @@
+This means that maybe the page won't always manage to get to a stable state before the screenshot is taken.
+
+###### Script B
+```xml
+<test>
+	[...]
+	<click css="#bigButton" timeout="5000"/>
+	<sleep duration="5000"/>
+	<screen/>
+</test>
+```
+The screenshot taken during this test captures how the page looks like five seconds after the #bigButton is clicked. 
+First the button is clicked if it appears within five seconds, then the test waits five seconds for the page to load 
+and eventually the screenshot is taken.
+
+#### Use timeouts
+
+Most commands for interacting with the elements of the page under test have timeout variables. 
+Setting a timeout means that the command will wait for an element to achieve the required state before the test will 
+try to interact with it.
+
+##### Example
+###### Script A
+```xml
+<test>
+	[...]
+	<click css="#bigButton" timeout="5000"/>
+	<screen/>
+</test>
+```
+The test doesn't click the button unless it's visible and clickable. The maximum waiting time is 5000ms, 
+after that the click command is ignored, an exception is thrown in the logs and the test continues.
+
+###### Script B
+```xml
+<test>
+	[...]
+	<click css="#bigButton"/>
+	<screen/>
+</test>
+```
+The test doesn't click the button unless it's visible and clickable. 
+The maximum waiting time is 1000ms because that's the default waiting time, after that the click command is ignored, 
+an exception is thrown in the logs and the test continues.
+
+###### Script C
+```xml
+<test>
+	[...]
+	<click css="#bigButton" timeout="0"/>
+	<screen/>
+</test>
+```
+The test doesn't click the button unless it's visible and clickable. The maximum waiting time is 0ms, 
+so unless the button is ready right away, the click command is ignored, 
+an exception is thrown in the logs and the test continues.
+
+#### Suplement wait commands with sleep commands
+
+Using wait commands and commands with timeouts is good for interacting with specific elements of the page under test. 
+Using the sleep command after that is even better, because it allows the page under test to load 
+without having the user write a wait command for each element on the page that might still be loading.
+
+##### Example
+```xml
+<test>
+	[...]
+	<click css="#bigButton" timeout="5000"/>
+	<sleep duration="5000"/>
+	<screen/>
+</test>
+```
+The screenshot taken during this test captures how the page looks like five seconds after the #bigButton is clicked. 
+First the button is clicked if it appears within five seconds, then the test waits five seconds for the page to load 
+and eventually the screenshot is taken.
+
+#### Don't overuse the sleep command
+
+Don't rely solely on sleep commands with huge durations. Using wait commands instead of sleep allows the tests 
+to run much faster in case if pages load properly and aren't broken.
+
+Examples
+
+###### Script A
+```xml
+<test>
+	[...]
+	<sleep duration="20000"/>
```

Comment:
Could we add some example on how to use `<wait-for-element-to-be-visible>` modifier instead of sleep  ?